SignalServiceKit/src/Messages/OWSIncompleteCallsJob.m

```diff
@@ -0,0 +1,150 @@
+//
+//  Copyright (c) 2018 Open Whisper Systems. All rights reserved.
+//
+
+#import "OWSIncompleteCallsJob.h"
+#import "OWSPrimaryStorage.h"
+#import "TSCall.h"
+#import <YapDatabase/YapDatabase.h>
+#import <YapDatabase/YapDatabaseQuery.h>
+#import <YapDatabase/YapDatabaseSecondaryIndex.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+static NSString *const OWSIncompleteCallsJobCallTypeColumn = @"call_type";
+static NSString *const OWSIncompleteCallsJobCallTypeIndex = @"index_calls_on_call_type";
+
+@interface OWSIncompleteCallsJob ()
+
+@property (nonatomic, readonly) OWSPrimaryStorage *primaryStorage;
+
+@end
+
+#pragma mark -
+
+@implementation OWSIncompleteCallsJob
+
+- (instancetype)initWithPrimaryStorage:(OWSPrimaryStorage *)primaryStorage
+{
+    self = [super init];
+    if (!self) {
+        return self;
+    }
+
+    _primaryStorage = primaryStorage;
+
+    return self;
+}
+
+- (NSArray<NSString *> *)fetchIncompleteCallIdsWithTransaction:(YapDatabaseReadWriteTransaction *)transaction
+{
+    OWSAssert(transaction);
+
+    NSMutableArray<NSString *> *messageIds = [NSMutableArray new];
+
+    NSString *formattedString = [NSString stringWithFormat:@"WHERE %@ == %d OR  %@ == %d",
+                                          OWSIncompleteCallsJobCallTypeColumn,
+                                          (int)RPRecentCallTypeOutgoingIncomplete,
+                                          OWSIncompleteCallsJobCallTypeColumn,
+                                          (int)RPRecentCallTypeIncomingIncomplete];
```

Comment:
This seems fine, but IIRC SQL has IN syntax so we could also do "WHERE call_type IN (incoming incomplete, outgoing incomplete)".  Didn't bother in this case, but if we were trying to match many enum values it'd probably be  worthwhile.